guide/src/migration.md

```diff
@@ -15,6 +15,47 @@ For projects embedding Python in Rust, PyO3 no longer automatically initalizes a
 
 The limitation of the new default implementation is that it cannot support multiple `#[pymethods]` blocks for the same `#[pyclass]`. If you need this functionality, you must enable the `multiple-pymethods` feature which will switch `#[pymethods]` to the inventory-based implementation.
 
+### Deprecated `#[pyproto]` methods
+
+Some methods such as `__bytes__` and `__format__` have been possible to implement two ways in PyO3 for some time: via a `#[pyproto]` (e.g. `PyBasicProtocol` for the methods listed here), or by writing them directly in `#[pymethods]`. This is only true for a handful of the `#[pyproto]` methods (for technical reasons to do with the Python C-API).
```

Comment:
Some protocol/dunder methods? Also, if we mention ` (for technical reasons to do with the Python C-API).`, we should also note that the situation can change in the future. `Technical reasons` can look like something unfixable for me.

Reply:
Good point. I changed to `for technical reasons to do with the way PyO3 currently interacts with the Python C-API`, which I think captures the fact that this might change.